### DISCUSSION TITLE
Text Prompt validation with another variable

### DISCUSSION THREAD

**Discussion by kevball2** [spectreconsole/spectre.console#633] [Q&A] [answered]
is it possible to validate a value of a text prompt based on another variable? 

``` csharp
var oldWage = 12.5
var newWage = AnsiConsole.Prompt(
                new TextPrompt<double>("Enter the workers new hourly wage")
                    .Validate(newWage =>
                    {
                        return newWage switch
                        {
                            < oldWage => ValidationResult.Error("[yellow]New wage cannot be less than current wage[/]"),
                            _ => ValidationResult.Success(),
                        };
                    }));
```

**Comment by nils-a** [comment, marked as answer]
I'm guessing your question is related to the error your code produces:

> CS0150: A constant value is expected

The reason for this is not really dependent on *spectre.console* but rather on the [C# specs for the `switch`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression).
You would need to change your switch to a [*`var` pattern*](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#var-pattern) like this:

```cs
return newWage switch
{
  var x when x < oldWage => ValidationResult.Error("[yellow]New wage cannot be less than current wage[/]"),
  _ => ValidationResult.Success(),
};
```

alternatively you can always use an [`if`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-if-statement) - which in this case would (in my opinion) be much, much clearer.

**Reply by kevball2** [reply]
Thanks @nils-a !  I will give that a go. Appreciate the background on the switch statement as well. New C# are still trying a handle on programming.
